### PR TITLE
decoupled openvpn-ldap-auth package from additional packages.

### PR DESCRIPTION
**Pull request by metallaw**

### CODE REVIEW

File: manifests/server.pp [hunk 1]
```diff
@@ -679,6 +679,9 @@
   }
 
   if $ldap_enabled == true {
+    package {$openvpn::params::ldap_auth_plugin_package:
```
Comment (bastelfreak) [inline manifests/server.pp]: can you add ldap_auth_plugin_package as additional param to the server class?

Comment (metallaw) [reply manifests/server.pp]: Hi, yeah sure. Thanks!

File: manifests/params.pp [hunk 2]
```diff
@@ -29,11 +29,13 @@
...
         $additional_packages = ['easy-rsa']
-        $ldap_auth_plugin_location = undef
+        $ldap_auth_plugin_package = ['openvpn-auth-ldap']
+        $ldap_auth_plugin_location = '/usr/lib64/openvpn/plugin/lib/openvpn-auth-ldap.so'
         $systemd = true
       # Redhat/Centos == 6.0
       } elsif(versioncmp($::operatingsystemrelease, '6.0') >= 0) and $::operatingsystem != 'Amazon' {
-        $additional_packages = ['easy-rsa','openvpn-auth-ldap']
+        $additional_packages = ['easy-rsa']
+        $ldap_auth_plugin_package = ['openvpn-auth-ldap']
```
Comment (luxflux) [inline manifests/params.pp]: Please also do this for the other OS-variants, for consistency.

File: manifests/params.pp [hunk 3]
```diff
@@ -29,11 +29,13 @@
       # Redhat/Centos >= 7.0
       if(versioncmp($::operatingsystemrelease, '7.0') >= 0) and $::operatingsystem != 'Amazon' {
         $additional_packages = ['easy-rsa']
-        $ldap_auth_plugin_location = undef
+        $ldap_auth_plugin_package = ['openvpn-auth-ldap']
```
Comment (bastelfreak) [inline manifests/params.pp]: does it need to be a an array or should we stick with a string? I think string is enough.

**Comment by bastelfreak** [comment]
Hi @metallaw , can you please rebase?

**Comment by bastelfreak** [comment]
@metallaw the git history looks a bit broken. Can you rebase it again? If you need any assistance feel free to join our IRC channel #voxpupuli on freenode.

**Comment by metallaw** [comment]
Rebased - But CI build fails since it expects the package openvpn-ldap-plugin to be installed on Debian 
=> is_expected.to contain_package('openvpn-auth-ldap') will need a rework of the openvpn_install_spec.rb

**Comment by jsnod** [comment]
I need this PR to use LDAP on CentOS 7 because in v7.2.0 of this module `server.pp` has `require => Package['openvpn-auth-ldap']` but the package is not installed for CentOS 7 due to `$additional_packages = ['easy-rsa']` (`openvpn-auth-ldap` is missing).  I'm pretty sure this PR will fix it -- any chance of landing this and making a release?

**Comment by bastelfreak** [comment]
Hi people. this branch looks pretty broken and inactive. I'm going to close it due to inactivity. Please reopen or resubmit it if you're still interested.

**Pull request closed without merging**